### PR TITLE
EPMLSTRCMW-187

### PR DESCRIPTION
**Pull request by Phoenix100101**

### CODE REVIEW

File: services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala [hunk 1]
```diff
@@ -386,7 +387,7 @@ class GitLabProjectVersioningTest
         }
       }
 
-      "throw new VersioningException" taggedAs Service in {
+      "throw new VersioningException" taggedAs(Service, ScheduledResource) in {
         val errorText = ""
         when(projectVersionRequest(withRepoProject)).thenReturn(
           Future.successful(
```
Comment (alesavin) [inline services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala]: Should we change tests at getFilesTree as well?

**Review by alesavin — COMMENTED** [review]
If we have to mark all tests inside GitLabProjectVersioningTest could we use "tag all tests in an entire class" feature http://doc.scalatest.org/3.0.1-2.12/org/scalatest/Tag.html ?

File: services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala [hunk 2]
```diff
@@ -53,7 +55,7 @@ class GitLabProjectVersioningTest
           payload = postProject
         )
 
-      "throw new VersioningException for inactive project" taggedAs Service in {
+      "throw new VersioningException for inactive project" taggedAs Service  in {
```
Comment (alesavin) [inline services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala]: I think it is better to remove this excessive space and have not change in line

**Comment by alesavin** [comment]
Travis failed, this is expected?

[error] (services / Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 73 s (01:13), completed Oct 30, 2020, 6:44:09 AM

**Comment by alesavin** [comment]
I see at travis run we still run

[info] GitLabProjectVersioningTest:
[info] GitLabProjectVersioning
[info]   when createRepository
[info]   - should throw new VersioningException for inactive project

We have to exclude test at travis job config by "-l" 
https://www.scalatest.org/user_guide/using_scalatest_with_sbt

**Review by myazinn — APPROVED** [review]
LGTM

File: services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
-import java.nio.file.{ Path, Paths }
+import java.nio.file.{Path, Paths}
 
 import akka.http.scaladsl.model.StatusCodes
 import cromwell.pipeline.datastorage.dao.repository.utils.TestProjectUtils
 import cromwell.pipeline.datastorage.dto.File.UpdateFileRequest
 import cromwell.pipeline.datastorage.dto._
-import cromwell.pipeline.utils.{ AkkaTestSources, ApplicationConfig, GitLabConfig, HttpStatusCodes }
+import cromwell.pipeline.utils.tags.annotations.ScheduledResource
+import cromwell.pipeline.utils.{AkkaTestSources, ApplicationConfig, GitLabConfig, HttpStatusCodes}
```
Comment (myazinn) [inline services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala]: Don't forget to reformat code. Scalafmt should have helped you with that.

File: utils/src/test/scala/cromwell/pipeline/utils/tags/ScheduledResource.scala [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package cromwell.pipeline.utils.tags
+
+import org.scalatest.Tag
+
+object ScheduledResource extends Tag("cromwell.pipeline.utils.tags.annotations.ScheduledResource")
```
Comment (myazinn) [inline utils/src/test/scala/cromwell/pipeline/utils/tags/ScheduledResource.scala:5]: I guess this is not required anymore.

**Review by myazinn — CHANGES_REQUESTED** [review]

**Comment by alesavin** [comment]
Let rebase after #120, hope all will be fine then

File: .travis.yml [hunk 5]
```diff
@@ -2,8 +2,11 @@ language: scala
 scala:
 - 2.12.9
 script:
-- sbt clean compile undeclaredCompileDependenciesTest coverage test coverageAggregate
-  scalastyle it:test
+- sbt clean compile
+- sbt undeclaredCompileDependenciesTest
+- sbt "testOnly -- -l cromwell.pipeline.utils.tags.annotaions.ScheduledResource"
```
Comment (myazinn) [inline .travis.yml]:
```suggestion
- sbt coverage "testOnly -- -l cromwell.pipeline.utils.tags.annotaions.ScheduledResource"
```

File: .travis.yml [hunk 6]
```diff
@@ -2,8 +2,11 @@ language: scala
...
+- sbt scalastyle
+- sbt coverage it:test
```
Comment (myazinn) [inline .travis.yml:9]:
```suggestion
- sbt coverage it:test
- sbt coverageAggregate
```

**Review by myazinn — COMMENTED** [review]
Something in this PR broke coverage checks. I'm not sure, but maybe the following suggestions will fix it.

File: .travis.yml [hunk 7]
```diff
@@ -2,8 +2,12 @@ language: scala
 scala:
 - 2.12.9
 script:
-- sbt clean compile undeclaredCompileDependenciesTest coverage test coverageAggregate
-  scalastyle it:test
+- sbt clean compile
+- sbt undeclaredCompileDependenciesTest
+- sbt coverage "testOnly -- -l cromwell.pipeline.utils.tags.annotaions.ScheduledResource"
```
Comment (myazinn) [inline .travis.yml:7]:
```suggestion
- sbt coverage "testOnly -- -l cromwell.pipeline.utils.tags.annotations.ScheduledResource"
```

File: README.md [hunk 8]
```diff
@@ -85,3 +85,34 @@ docker-compose -f postgres-mongo.yml up
 
 ## Developer Guide
 https://kb.epam.com/display/EPMLSTR/Cromwell+Developer+Guide
+
+##Testing
+
+ScheduledResource this is tag for tests working with scheduled resources.
+If you want to skip this test you can run this command:
+```
+testOnly -- -l cromwell.pipeline.utils.tags.annotaions.ScheduledResource
```
Comment (myazinn) [inline README.md:94]:
```suggestion
testOnly -- -l cromwell.pipeline.utils.tags.annotations.ScheduledResource
```

**Review by myazinn — CHANGES_REQUESTED** [review]
All checks are back again which is great! However, I've noticed that tests from `GitLabProjectVersioningTest` are still run on Travis. This is probably due to a typo.

**Comment by iliaonishchenko** [comment]
This PR is out of date. Closing it.

**Pull request closed without merging**